test/solarwinds_apm/config_test.rb

```diff
@@ -0,0 +1,274 @@
+      ENV['DUMMY_KEY'] = 'false'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong env, use default true ' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with wrong env, use default true ' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with config env true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with config env false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, false, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, config env true, use default false, should be true' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, empty config env, use default false, should be false' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, '', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with nil env, empty config env, use default true, should be true' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, '', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+  end
+
+  describe '#set_log_level' do
+    before do
+      SolarWindsAPM::Config.initialize
+      @current_debug_level = ENV['SW_APM_DEBUG_LEVEL']
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+    end
+
+    after do
+      ENV['SW_APM_DEBUG_LEVEL'] = @current_debug_level
+    end
+    
+    it 'debug_level is out of range use default 1' do
+      SolarWindsAPM::Config[:debug_level] = 7
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 1
+    end
+
+    it 'debug_level is in the range' do
+      SolarWindsAPM::Config[:debug_level] = 1
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 3
```

Comment:
Hmm, why is the log level 3 if we're setting what looks to be a valid level of 1?
Also just noticed https://github.com/solarwinds/apm-ruby/blob/65c5a43cf8bac1a7de2ff59fc95d772575c2430c/lib/solarwinds_apm/config.rb#L71 has valid levels -1 through 6, while our [customer doc](https://github.com/solarwinds/apm-ruby/blob/65c5a43cf8bac1a7de2ff59fc95d772575c2430c/CONFIGURATION.md#reference) has 0 through 6--which is correct?

Reply:
I think customer doc is correct.